kasa/tests/newfakes.py

```diff
@@ -306,26 +305,34 @@ def __init__(self, info):
         )
         self.info = info
 
-    @property
-    def needs_handshake(self) -> bool:
-        return False
-
-    @property
-    def needs_login(self) -> bool:
-        return False
-
-    async def login(self, request: str) -> None:
-        pass
-
-    async def handshake(self) -> None:
-        pass
-
     async def send(self, request: str):
         request_dict = json_loads(request)
+        method = request_dict["method"]
+        params = request_dict["params"]
+        if method == "multipleRequest":
+            responses = []
+            for request in params["requests"]:
+                response = self._send_request(request)  # type: ignore[arg-type]
+                response["method"] = request["method"]  # type: ignore[index]
+                responses.append(response)
+            return {"result": {"responses": responses}, "error_code": 0}
+        else:
+            return self._send_request(request_dict)
+
+    def _send_request(self, request_dict: dict):
         method = request_dict["method"]
         params = request_dict["params"]
         if method == "component_nego" or method[:4] == "get_":
-            return {"result": self.info[method], "error_code": 0}
+            if method in self.info:
+                return {"result": self.info[method], "error_code": 0}
+            else:
+                warnings.warn(
+                    UserWarning(
+                        f"Fixture missing expected method {method}, try to regenerate"
+                    ),
+                    stacklevel=1,
+                )
+                return {"result": {}, "error_code": 0}
```

Comment:
Perhaps we should return a non-success here?

Reply:
This is a condition that will only happen when we start adding new queries and older fixtures have not picked them up yet.  In that instance the current plan is to fall back to a base proto type implementation like the `FakeTransportProtocol` although I'm hopeful to nail down all of the associated queries resulting from `component_nego` results so hopefully we can avoid this altogether.  The trouble with an error is that the tests would then fail.  Alternatively I could try to do a clever `pytest.skip` but I haven't figured out how to do that yet.

Reply:
Aah, I see. This is fine, thanks for the explanation :-)